apps/lbmMultiRes/lbmMultiRes.cu

```diff
@@ -0,0 +1,781 @@
+                            const Neon::domain::mGrid::Field<CellType>& cellType,
+                            const Neon::domain::mGrid::Field<T>&        postCollision,
+                            Neon::domain::mGrid::Field<T>&              postStreaming)
+{
+    //regular Streaming of the normal voxels at level L which are not interfaced with L+1 and L-1 levels.
+    //This is "pull" stream
+
+    return grid.getContainer(
+        "stream_" + std::to_string(level), level,
+        [&, level](Neon::set::Loader& loader) {
+            const auto& type = cellType.load(loader, level, Neon::MultiResCompute::STENCIL);
+            const auto& fpost_col = postCollision.load(loader, level, Neon::MultiResCompute::STENCIL);
+            auto        fpost_stm = postStreaming.load(loader, level, Neon::MultiResCompute::MAP);
+
+            return [=] NEON_CUDA_HOST_DEVICE(const typename Neon::domain::bGrid::Cell& cell) mutable {
+                if (type(cell, 0) == CellType::bulk) {
+                    //If this cell has children i.e., it is been refined, than we should not work on it
+                    //because this cell is only there to allow query and not to operate on
+                    if (!fpost_stm.hasChildren(cell)) {
+
+                        for (int8_t q = 0; q < Q; ++q) {
+                            const Neon::int8_3d dir = -getDir(q);
+
+                            //if the neighbor cell has children, then this 'cell' is interfacing with L-1 (fine) along q direction
+                            if (!fpost_stm.hasChildren(cell, dir)) {
+                                auto nghType = type.nghVal(cell, dir, 0, CellType::undefined);
+
+                                if (nghType.isValid) {
+                                    if (nghType.value == CellType::bulk) {
+                                        fpost_stm(cell, q) = fpost_col.nghVal(cell, dir, q, T(0)).value;
+                                    } else {
+                                        const int8_t opposte_q = latticeOppositeID[q];
+                                        fpost_stm(cell, q) = fpost_col(cell, opposte_q) + fpost_col.nghVal(cell, dir, opposte_q, T(0)).value;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+        });
+}
+
+template <typename T, int Q>
+Neon::set::Container explosionPull(Neon::domain::mGrid&                 grid,
+                                   int                                  level,
+                                   const Neon::domain::mGrid::Field<T>& postCollision,
+                                   Neon::domain::mGrid::Field<T>&       postStreaming)
+{
+    // Initiated by the fine level (hence "pull"), this function performs a coarse (level+1) to
+    // fine (level) communication or "explosion" by simply distributing copies of coarse grid onto the fine grid.
```

Comment:
I'd suggest changing the ordering schema for the levels to match the one used in tree structures, where the toot is a the level 0.